### PR TITLE
Feature/sendmsg

### PR DESCRIPTION
**Pull request by fabriziobertocci**
Added implementation and unit test for sendmsg() and recvmsg().
Tested on Linux, Mac and Windows.

### CODE REVIEW

File: test/libc/sock/sendrecvmsg_test.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/*-*- mode:c;indent-tabs-mode:nil;c-basic-offset:2;tab-width:8;coding:utf-8 -*-│
+│vi: set net ft=c ts=2 sts=2 sw=2 fenc=utf-8                                :vi│
+╞══════════════════════════════════════════════════════════════════════════════╡
+│ Copyright 2021                                                               │
```
Comment (jart) [inline test/libc/sock/sendrecvmsg_test.c:4]: Please edit this line.

File: test/libc/sock/sendrecvmsg_test.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  ASSERT_NE(-1, socketpair(AF_UNIX, SOCK_STREAM, 0, fd));
+  ASSERT_EQ(hwLen, sendmsg(fd[0], &msg, 0));
+  
+  data[0].iov_base = malloc(20);
+  data[0].iov_len = 20;
+  ASSERT_NE(NULL, data[0].iov_base);
+  msg.msg_iovlen = 1;
+  ASSERT_EQ(hwLen, recvmsg(fd[1], &msg, 0));
+
+  ASSERT_EQ(0, memcmp(&(((char *)(msg.msg_iov[0].iov_base))[0]), hello, strlen(hello)));
```
Comment (jart) [inline test/libc/sock/sendrecvmsg_test.c:55]: Try changing `malloc(20)` to `gc(xcalloc(20))`. In that case, you can delete the null check and these two lines can simply be `EXPECT_STREQ("helloworld", msg.msg_iov[0].iov_base);`. You'll need these too of course:

```c
#include "libc/runtime/gc.internal.h"
#include "libc/x/x.h"
```

**Review by jart — CHANGES_REQUESTED** [review]
Looks good, with two minor nits. Very happy to see the unit test. Thank you for sending this!

File: test/libc/sock/sendrecvmsg_test.c (hunk 1 above)
Comment (fabriziobertocci) [reply test/libc/sock/sendrecvmsg_test.c:4]: Hmm I copied the header from the socketpair test tha Allison wrote... I removed her name but I forgot to add your...
changed.

**Review by jart — APPROVED** [review]
LGTM

**Pull request merged**